src/test/unit/PaymentCoordinatorUnit.t.sol

```diff
@@ -0,0 +1,1613 @@
+        assertEq(claimer, paymentCoordinator.claimerFor(earner), "claimerFor not set");
+        cheats.stopPrank();
+    }
+
+    function testFuzz_setCalculationIntervalSeconds(uint64 intervalSeconds) public {
+        cheats.startPrank(paymentCoordinator.owner());
+        cheats.expectEmit(true, true, true, true, address(paymentCoordinator));
+        emit CalculationIntervalSecondsSet(paymentCoordinator.calculationIntervalSeconds(), intervalSeconds);
+        paymentCoordinator.setCalculationIntervalSeconds(intervalSeconds);
+        assertEq(
+            intervalSeconds,
+            paymentCoordinator.calculationIntervalSeconds(),
+            "calculationIntervalSeconds not set"
+        );
+        cheats.stopPrank();
+    }
+
+    function testFuzz_setRetroactivePaymentsEnabled(bool retroactivePaymentsEnabled) public {
+        cheats.prank(paymentCoordinator.owner());
+        paymentCoordinator.setRetroactivePaymentsEnabled(retroactivePaymentsEnabled);
+        assertEq(
+            retroactivePaymentsEnabled,
+            paymentCoordinator.retroactivePaymentsEnabled(),
+            "retroactivePaymentsEnabled not set"
+        );
+    }
+
+    function testFuzz_setActivationDelay(uint64 activationDelay) public {
+        cheats.startPrank(paymentCoordinator.owner());
+        cheats.expectEmit(true, true, true, true, address(paymentCoordinator));
+        emit ActivationDelaySet(paymentCoordinator.activationDelay(), activationDelay);
+        paymentCoordinator.setActivationDelay(activationDelay);
+        assertEq(activationDelay, paymentCoordinator.activationDelay(), "activationDelay not set");
+        cheats.stopPrank();
+    }
+
+    function testFuzz_setGlobalOperatorCommission(uint16 globalCommissionBips) public {
+        cheats.startPrank(paymentCoordinator.owner());
+        cheats.expectEmit(true, true, true, true, address(paymentCoordinator));
+        emit GlobalCommissionBipsSet(paymentCoordinator.globalOperatorCommissionBips(), globalCommissionBips);
+        paymentCoordinator.setGlobalOperatorCommission(globalCommissionBips);
+        assertEq(
+            globalCommissionBips,
+            paymentCoordinator.globalOperatorCommissionBips(),
+            "globalOperatorCommissionBips not set"
+        );
+        cheats.stopPrank();
+    }
+
+    function testFuzz_setPaymentUpdater(address newPaymentUpdater) public {
+        cheats.startPrank(paymentCoordinator.owner());
+        cheats.expectEmit(true, true, true, true, address(paymentCoordinator));
+        emit PaymentUpdaterSet(paymentCoordinator.paymentUpdater(), newPaymentUpdater);
+        paymentCoordinator.setPaymentUpdater(newPaymentUpdater);
+        assertEq(newPaymentUpdater, paymentCoordinator.paymentUpdater(), "paymentUpdater not set");
+        cheats.stopPrank();
+    }
+
+    function testFuzz_setPayAllForRangeSubmitter(address submitter, bool newValue) public {
+        cheats.startPrank(paymentCoordinator.owner());
```

Comment:
would be good to have sanity testes for "NotOwnerCannotXYZ"

Reply:
added